### ISSUE TITLE
Support Dask Dataframe Transpose

### ISSUE THREAD

**Issue by mohamedniyaz1996** [dask/dask#8357]
Please add API-support for DataFrame transpose as we do in pandas with DataFrame.transpose()

**Comment by quasiben** [comment]
Something like the following might work. 

> dd.from_dask_array(ddf.to_dask_array().T.compute_chunk_sizes()).compute()

Though many things probably need to be better taken care of like columns to index for example.

**Comment by jsignell** [comment]
Yeah I think this could be a good feature and it could be built using blockwise, but it might take a little thought to get right. I tried a naive approach and it didn't do quite the right thing.

```python
import pandas as pd
import dask.dataframe as dd

df = pd.DataFrame(data={'col1': [1, 2, 4, 5], 'col2': [3, 4, 6, 9]})
ddf = dd.from_pandas(df, 2)
ddf.map_partitions(lambda df: df.transpose(), enforce_metadata=False).compute()
```
```python
        0    1    2    3
col1  1.0  2.0  NaN  NaN
col2  3.0  4.0  NaN  NaN
col1  NaN  NaN  4.0  5.0
col2  NaN  NaN  6.0  9.0
```

I'll just also say that since dataframe partitions the data along the index, the data would have to be entirely repartitioned if you want to know the divisions (range of each partition).
